

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Applications &mdash; Carousel 1.0 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Using Carousel" href="using_carousel.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >
          

          
            <a href="index.html" class="icon icon-home"> Carousel
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="using_carousel.html">Using Carousel</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Applications</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#r-markdown">R markdown</a></li>
</ul>
</li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Carousel</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Applications</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/apps.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="applications">
<h1>Applications<a class="headerlink" href="#applications" title="Permalink to this headline">¶</a></h1>
<p><em>Carousel users have access to a variety of applications. Simply upload your data, launch your analysis, and share with your collaborators!</em></p>
<div class="section" id="r-markdown">
<h2>R markdown<a class="headerlink" href="#r-markdown" title="Permalink to this headline">¶</a></h2>
<p>Render an <a class="reference external" href="https://rmarkdown.rstudio.com/">R Markdown document</a> using custom code and private data.
If the R Markdown document is being used to display data which is contained in other data files,
just upload those additional files and reference them directly in your R Markdown code.</p>
<blockquote>
<div><p>Custom R/Shiny</p>
</div></blockquote>
<hr class="docutils" />
<p>Render an <a class="reference external" href="https://shiny.rstudio.com/">R/Shiny dashboard</a> using custom code and private data.
Using this app, a user may render any sort of R/Shiny dashboard, including the use of other data files for importing data for visualization.
The R/Shiny app in Carousel supports many common R libraries for data manipulation and visualization, but please
<a class="reference external" href="mailto:hutchddatacore&#37;&#52;&#48;fredhutch&#46;org">be in touch</a> if you require additional libraries for your visualization.</p>
<blockquote>
<div></div></blockquote>
<p>The <a class="reference external" href="https://merenlab.org/software/anvio/">anvi’o</a> software library was created by the Meren Lab to provide
an open-source, community-driven analysis and visualization platform for microbial ‘omics data.
This particular Carousel app is used to compare a set of microbial genomes using the anvi’o pangenome display.
To prepare a set of genomes for display with this app, first pre-processs them with the
<a class="reference external" href="https://github.com/FredHutch/nf-anvio-pangenome">nf-anvio-pangenome workflow</a>.
The two anvi’o database files created by that workflow can be directly uploaded to Carousel and visualized using this app.
The extensive anvi’o documentation produced by the Meren Lab includes an
<a class="reference external" href="https://merenlab.org/2016/11/08/pangenomics-v2/">explanatory tutorial on pan-genome analysis</a> which may be helpful
for understanding the display provided by this app.</p>
<blockquote>
<div></div></blockquote>
<p><a class="reference external" href="https://jbrowse.org/jb2">JBrowse2</a> is a pluggable open-source platform for visualizing and integrating biological data.
JBrowse2’s most important feature is that it is able to compose multiple views of your data together on the same screen.
The JBrowse2 <a class="reference external" href="https://jbrowse.org/jb2/">website</a> contains extensive documentation, demos, and educational resources.
In order to display your data, JBrowse 2 requires (1) the reference genome for your organism of interest and
(2) a set of tracks that overlay additional information on that genome.</p>
<blockquote>
<div></div></blockquote>
<p>The Chan-Zuckerberg Institute has created an interactive data explorer for single-cell transcriptomics called
<a class="reference external" href="https://github.com/chanzuckerberg/cellxgene">cellxgene</a> (pronounced “cell-by-gene”). The cellxgene
<a class="reference external" href="https://github.com/chanzuckerberg/cellxgene">website</a> contains documentation on how to prepare your data for display.
Once you have prepared the .h5ad file required by cellxgene, upload it to Carousel for display using this app.</p>
<blockquote>
<div></div></blockquote>
<p>The GLAM (Gene-Level Analysis of Metagenomes) Browser was created to allow scientists to interactively
explore the results of gene-level metagenomics analysis performed using the <a class="reference external" href="https://github.com/Golob-Minot/geneshot/wiki">geneshot</a>
bioinformatics workflow for microbiome analysis.
After running <a class="reference external" href="https://github.com/Golob-Minot/geneshot/wiki">geneshot</a> (v0.9 or later), a file will be created in the output directory ending in <cite>.rdb</cite>.
This file contains a compressed version of the output data which has been optimized for rapid visualization using the GLAM Browser.
Upload the <cite>.rdb</cite> file for any experiment to Carousel using this app, and you will be able to inspect the characteristics of every detected organism,
as well as their association with experimental parameters.
Selecting any individual gene group will show further details, including the taxonomic and functional assignments (if any) for the genes in that group.</p>
<blockquote>
<div></div></blockquote>
<p>When analyzing a collection of bacterial genomes, it can be useful to summarize which of the genomes contain a particular set of genes.
In particular, it can be useful to have a display which summarizes both (a) how similar the genomes are to each other alongside
(b) whether they contain a particular set of genes.The analysis pipeline which can be used to generate this information is called
BOFFO (Bacterial Operon Finder for Functional Organization). The inputs to this pipeline are (1) a list of genomes to analyze, and
(2) a collection of genes to align against those genomes. After running the BOFFO pipeline, this app can be used to visualize the
results in a single display which combines a genome similarity dendrogram with a table showing which genes are present.
In addition, this app provides the ability to customize the visual display and export a vector PDF which can be used to generate publication-quality figures.</p>
<blockquote>
<div></div></blockquote>
<p>Sometimes you just want to make a figure. One of the most widely used software libraries for making images out of data is Plotly.
Using a table of data as an input, this software can produce a large variety of displays. To make some of this functionality
available without needing to use the command line, this app will render data provided as a spreadsheet (in CSV format) using
the Plotly library. Simply select the columns which contain the data corresponding to the X-axis, Y-axis, etc., and the app
will display an interactive figure which the user can customize and even download.</p>
<blockquote>
<div></div></blockquote>
<p>We are always looking to add new visualizations and analyses to Carousel.
Any open-source visualization tool that is compatible with Docker can be added to Carousel.</p>
<p>Put more simply the tool needs to be:</p>
<ul class="simple">
<li><p>Free</p></li>
<li><p>Able to run on Linux</p></li>
<li><p>Have an interactive visualization component</p></li>
</ul>
<p>If you are unsure if a tool is right for Carousel don’t hesitate to <a class="reference external" href="mailto:hutchdatacore&#37;&#52;&#48;fredhutch&#46;org">contact us!</a></p>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="using_carousel.html" class="btn btn-neutral float-left" title="Using Carousel" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, L. Wolfe, S. Minot, D. Nambi.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
    <!-- Theme Analytics -->
    <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-XXXXXXX-1', 'auto');
    
    ga('send', 'pageview');
    </script>

    
   

</body>
</html>